src/c2goto/library/pthread_lib.c

```diff
@@ -706,3 +711,26 @@ __ESBMC_HIDE:;
   // This function always succeeds.
   return res;
 }
+
+/************************ detach routine ***********************/
+
+// The pthread_detach() function marks the thread identified by thread
+// as detached.  When a detached thread terminates, its resources are
+// automatically released back to the system without the need for
+// another thread to join with the terminated thread.
+// source: https://man7.org/linux/man-pages/man3/pthread_detach.3.html
+int pthread_detach(pthread_t threadid)
+{
+__ESBMC_HIDE:;
+  __ESBMC_atomic_begin();
+  __ESBMC_assert(
+    !__ESBMC_pthread_thread_detach[(int)threadid],
```

Comment:
Just add a minor comment about the EINVAL return is being checked here

Reply:
Fixed.